### ISSUE TITLE
Continuous Deployment Job runs on forks

### ISSUE THREAD

**Issue by AndrewADev** [thebjorn/pydeps#114]
Hello,

I recently pulled updates from the main repo that I pushed to my (forked) remote, and ended up with a [failed Workflow run](https://github.com/AndrewADev/pydeps/runs/3644331180?check_suite_focus=true) because it apparently ran the CD job, which includes talking to PyPI when run on the [default branch](https://github.com/thebjorn/pydeps/blob/master/.github/workflows/ci-cd.yml#L60), regardless of repo (original or fork):
![image](https://user-images.githubusercontent.com/2937540/133923920-dbb39828-3dad-42b5-8a79-012ac03c8499.png)

The [`cd` job](https://github.com/thebjorn/pydeps/blob/master/.github/workflows/ci-cd.yml#L174) of `ci-cd.yml` should probably only be allowed to run for the main repo.

Note: It does seem that that there is a way for me to [disable an entire workflow](https://docs.github.com/en/actions/managing-workflow-runs/disabling-and-enabling-a-workflow), but seeing as the call to PyPI is a context more granular than that (a _job_), disabling at that level would also disable the CI job, which includes tests (likely something that would still be worth running on forks).

A solution might involve one of these approaches, or a combination thereof:
- Split the CI & CD jobs into separate workflows, thus allowing them to be enabled/disabled independently of one another
- Adding something like `github.repository == thebjorn/pydeps` to [line 175](https://github.com/thebjorn/pydeps/blob/master/.github/workflows/ci-cd.yml#L175) (job-level `if` conditional using the [`github` context object](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context)).

If one of those sounds good, let me know, and I can submit a PR.

**Comment by thebjorn** [comment]
It's nice to have the CD job run only after a successful CI, so the second option seems more appealing to me.

I'm always happy for PRs ;-)

I'm mentioning @aleks-ivanov since this might be of general interest to the Pipeline Project.

**Comment by aleks-ivanov** [comment]
@thebjorn thank you for the mention!

This a scenario where a merge from upstream that includes the new version is done in a fork's default branch, is one we haven't thought of and @AndrewADev 's suggestion is a viable option. Good catch @AndrewADev!

We'd submit a PR for this promptly 🙂
